### ISSUE TITLE
Update to memories 7.1.0 stuck on PostgresSQL caused by migration.

### ISSUE THREAD

**Issue by alexdrl** [pulsejet/memories#1107]
**Describe the bug**
Update to memories 7.1.0 stuck on PostgresSQL caused by migration SQL.
my database had 200k rows on oc_memories and 2000k on oc_filecache.

**To Reproduce**
Update to version 7.1.0 was really slow on my machine (not sure if Postgres could be the one to blame), connecting to pgadmin I found this SQL was stuck in executing.

```sql
  UPDATE oc_memories AS m
                    SET parent = f.parent
                    FROM oc_filecache AS f
                    WHERE f.fileid = m.fileid
```
                                       
The content on the SQL migration is similar [here](https://github.com/pulsejet/memories/blob/5a9908e3ae78ed1bf2bb0e32cffaa509bb255a9c/lib/Migration/Version800000Date20240329060325.php#L91C1-L97C19)

Could a join here be helpful for performance? Could we maybe filter similar to this?:

```sql
  UPDATE oc_memories AS m
                    SET parent = f.parent
                    FROM oc_filecache AS f
                    WHERE f.fileid = m.fileid AND m.parent = 0
```

Or maybe that is a dumb idea because the whole UPDATE is transactional?

**Screenshots**

**Platform:**

- OS: Windows
- Browser: MS Edge
- Memories Version: 7.0.2 to 7.1.0
- Nextcloud Version: [28.0.4
- PHP Version: 8.2
- Postgres 16.1

**Additional context**

- Any errors in the JS console?
- Any errors in the Nextcloud server logs?

PS: Thanks for the work on the memories app for nextcloud, I really really love it 💕

**Comment by pulsejet** [comment]
200k is actually tiny so I'm surprised it's slow. Were you upgrading from the command line? This is a one time thing anyway, so I don't see much advantage in optimizing here.

**Comment by pulsejet** [comment]
> Could we maybe filter similar to this?:

The filter won't do anything because `parent` _is_ zero on all rows.

**Comment by alexdrl** [comment]
> 200k is actually tiny so I'm surprised it's slow. Were you upgrading from the command line? This is a one time thing anyway, so I don't see much advantage in optimizing here.

yes, I was updating from occ upgrade command as updating the app from the Web UI got stuck too, but I didn't see any additional logs on the CLI. It might by my Postgres performance as runs with a NFS backend (but the backend is a powerful NAS server with ZFS), so unless I try to re-run the SQL query, I don't know... For getting out the issue, I checked and as the parent column is already filled, I created the migration row on the oc_migrations table... 

Actually I can run the raw SQL query again to rule out a random performance concern.

**Comment by pulsejet** [comment]
Yeah, 15 minutes is too much; something could be wrong with your db (possibly also causing other slowdown issues). It took me <5s on a similarly sized DB.

**Comment by alexdrl** [comment]
> Yeah, 15 minutes is too much; something could be wrong with your db (possibly also causing other slowdown issues). It took me <5s on a similarly sized DB.

It keeps on running 😅  
![image](https://github.com/pulsejet/memories/assets/26382633/064753c7-af75-4b52-9664-a3e02f40a9c5)

I remember that the update using the UI got a timeout, but maybe later the migration was kept in the background and eventually finished? adding this kind of filter could help with a migration retry in those cases 😄 

![image](https://github.com/pulsejet/memories/assets/26382633/7e8e6ef3-9081-4ab3-bc9b-82914e445bd4)

I will try to get some pgbench run on the server (not a Postgres expert btw), just to see if there is something very odd there.

**Comment by pulsejet** [comment]
> adding this kind of filter could help with a migration retry in those cases

Ah I didn't realise you retried the migration. That's a good point. It'll still scan everything though cause there's no index (needed) on parent.

I'm still not sure if this is worth the optimization though; it's a one-time thing. Let's see if someone else comes up with the same issue.

**Comment by pulsejet** [comment]
Definitely do check your DB. I just set all `parent` to `0` and re-ran the update query; it takes ~1s for 40k for me

![image](https://github.com/pulsejet/memories/assets/10709794/f656ec80-99e6-48e4-b6d1-8e627bb19eb0)

**Comment by alexdrl** [comment]
> Definitely do check your DB. I just set all `parent` to `0` and re-ran the update query; it takes ~1s for 40k for me (I'm running MariaDB though)
> 
> ![image](https://private-user-images.githubusercontent.com/10709794/318851425-f656ec80-99e6-48e4-b6d1-8e627bb19eb0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxMjc4MzEsIm5iZiI6MTcxMjEyNzUzMSwicGF0aCI6Ii8xMDcwOTc5NC8zMTg4NTE0MjUtZjY1NmVjODAtOTllNi00OGU0LWI2ZDEtOGU2MjdiYjE5ZWIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDA2NTg1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1MzdkNzYwOWUwMWZhOTA4NjJkYTA5ODA1Y2VjMzBjZDFkMzYyN2U0Njg4ZGQ2NDY4ZDNmOTZjMTg2YjQ5MzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OqrsE4aoeOH1TvpN9U9GkwG3nrgAAq9pGlV3kteUw3s)
> 
> Subsequently
> 
> ![image](https://private-user-images.githubusercontent.com/10709794/318851628-3292bfaa-8c14-4ea9-961c-0de612b7093e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxMjc4MzEsIm5iZiI6MTcxMjEyNzUzMSwicGF0aCI6Ii8xMDcwOTc5NC8zMTg4NTE2MjgtMzI5MmJmYWEtOGMxNC00ZWE5LTk2MWMtMGRlNjEyYjcwOTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDA2NTg1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2MzE1ZDE4YzMzZTFhNmJhM2RjMTVmZDFiNWEyOTM1ZGQwZWFmZmJiZGNhZTYwMzk1MzgxMmE3NjY5M2RmOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._v0DAfyH54cqP2D19W8fKuxpOuPiHbktQPl6y7f56WE)

100% agree in that it can be my Postgres performance, just raising my hand in case it can be a issue that could happen to somebody else...

Will try to get a pgbench done today and post the results for comparison. (not sure if that would be a meaningful metric though)
